be/src/runtime/routine_load/data_consumer.cpp

```diff
@@ -290,11 +290,17 @@ Status KafkaDataConsumer::get_partition_offset(std::vector<int32_t>* partition_i
     _last_visit_time = time(nullptr);
     beginning_offsets->reserve(partition_ids->size());
     latest_offsets->reserve(partition_ids->size());
+    MonotonicStopWatch watch;
+    watch.start();
     for (auto p_id : *partition_ids) {
         int64_t beginning_offset;
         int64_t latest_offset;
+        auto left_ms = timeout - watch.elapsed_time() / 1000 / 1000;
+        if (left_ms <= 0) {
```

Comment:
How about adding a LOG(WARNING) here?

Reply:
> How about adding a LOG(WARNING) here?

If this fails, the upper caller will print a warning log